x-pack/plugins/security_solution/public/detections/components/rules/rule_preview/index.tsx

```diff
@@ -123,6 +125,13 @@ const RulePreviewComponent: React.FC<RulePreviewProps> = ({
     setTimeFrame(defaultTimeRange);
   }, [ruleType]);
 
+  const { startTransaction } = useStartTransaction();
+
+  const handlePreviewClick = useCallback(() => {
+    startTransaction({ name: SINGLE_RULE_ACTIONS.PREVIEW });
+    createPreview();
+  }, [createPreview, startTransaction]);
```

Comment:
I'd like to chat about where does `startTransaction` call belong to -- to the component or to the logic that implements this action (`createPreview` function in this particular case).